src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -133,91 +126,242 @@ public void Add([NotNull] Operation operation, [NotNull] object objectToApplyTo)
             }
             else
             {
-                positionAsInteger = GetNumericEnd(path);
+                var checkNumericEndResult = GetNumericEndResult(path);
```

Comment:
My thought about this... this code could be simplified by doing one of two things:
1. Return an `int?` here - the extra type you created doesn't really do anything that `int?` doesn't.
2. Instead of 1 - do the 'split' and return that if it's a numeric ending-path. Basically roll all of the code in the 'if' into the `GetNumericEndResult` method